applications/ALEapplication/python_scripts/ale_analysis.py

```diff
@@ -0,0 +1,283 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing Kratos
+import KratosMultiphysics
+import KratosMultiphysics.ALEApplication as KratosALE
+
+# Importing the solvers (if available)
+try:
+    import KratosMultiphysics.ExternalSolversApplication
+    KratosMultiphysics.Logger.PrintInfo("ExternalSolversApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("ExternalSolversApplication", "not imported")
+try:
+    import KratosMultiphysics.EigenSolversApplication
+    KratosMultiphysics.Logger.PrintInfo("EigenSolversApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("EigenSolversApplication", "not imported")
+
+
+class ALEAnalysis(object): # TODO in the future this could derive from a BaseClass in the Core
+    """
+    This class is the main-script of the ALEapplication put in a class
+
+    It can be imported and used as "black-box"
+    """
+    def __init__(self, ProjectParameters, external_model_part=None):
+        if (type(ProjectParameters) != KratosMultiphysics.Parameters):
+            raise Exception("Input is expected to be provided as a Kratos Parameters object")
+        self.ProjectParameters = ProjectParameters
+        self.__CreateSolver(external_model_part)
+
+    #### Public functions to run the Analysis ####
+    def Run(self):
+        self.Initialize()
+        self.RunMainTemporalLoop()
+        self.Finalize()
+
+    def RunMainTemporalLoop(self):
+        while self.time < self.end_time:
+            self.InitializeTimeStep()
+            self.SolveTimeStep()
+            self.FinalizeTimeStep()
+
+    #### Public functions defining the Interface to the CoSimulationApplication ####
+    def Initialize(self):
+        self.__ExecuteInitialize()
+        self.__InitializeIO()
+        self.__ExecuteBeforeSolutionLoop()
+
+    def InitializeTimeStep(self):
+        self.__ExecuteInitializeSolutionStep()
+
+    def SolveTimeStep(self):
+        self.__SolveSolutionStep()
+
+    def FinalizeTimeStep(self):
+        self.__ExecuteFinalizeSolutionStep()
+
+    def Finalize(self):
+        self.__ExecuteFinalize()
+
+    #### Internal functions ####
+    def __CreateSolver(self, external_model_part=None):
+        """ Create the Solver (and create and import the ModelPart if it is not passed from outside) """
```

Comment:
I think the creation of the model part should be in a separate function. However I don't have a better suggestion since the creation of the model part is tightly coupled to the solver (via add variables).